### PR TITLE
Add getContent hook

### PR DESCRIPTION
**Pull request by dustinfarris**
Allow subclasses to override how content is generated, and make it async, e.g.:

``` js
MyPlugin.prototype.getContent = function() {
  return new RSVP.Promise(function(resolve) {
    resolve('something else');
  });
}
```

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -38,5 +47,12 @@ Creator.prototype.build = function () {
 
   mkdirp.sync(path.dirname(outputFilePath));
 
-  fs.writeFileSync(outputFilePath, this.content, this.fileOptions);
+  var _this = this;
+
+  return new RSVP.Promise(function(resolve) {
+    _this.getContent().then(function(content) {
```
Comment (stefanpenner) [inline index.js]: please resolve this expression. Correct promise usage would be something like:

``` js
var plugin = this;
return new RSVP.Promise(function(resolve) {
   resolve(plugin.getContent());
}).then(function(content) {
  fs.writeFileSync(....
});
```

File: index.js [hunk 2]
```diff
@@ -38,5 +47,12 @@ Creator.prototype.build = function () {
...
+      fs.writeFileSync(outputFilePath, content, _this.fileOptions);
+      resolve();
```
Comment (stefanpenner) [inline index.js]: this does not deal with the failure scenarios, see above comment

**Comment by dustinfarris** [comment]
Anyone available to look at this??  ping @rwjblue

**Comment by dustinfarris** [comment]
Can someone review this PR?

@rwjblue @stefanpenner

**Comment by stefanpenner** [comment]
> Can someone review this PR?

Sorry, I forgot to press "submit review"

**Comment by dustinfarris** [comment]
@stefanpenner thanks for taking a look at this.

I've updated the promise resolve code to match your example.  (looks and feels much better, thank you)

**Comment by stefanpenner** [comment]
@dustinfarris implementation  LGTM.

Some things:
- will need readme update
- we should confirm with @rwjblue re: the name of the function.

**Comment by stefanpenner** [comment]
@dustinfarris I believe I have added a feature, which supersedes this: https://github.com/rwjblue/broccoli-file-creator/pull/12. If you have time, could you take a look and confirm?

**Pull request closed without merging**